src/sqlfluff/core/config/file.py

```diff
@@ -0,0 +1,133 @@
+"""Lower level routines for config file loading and caching.
+
+Functions in this module load config from *individual* files and
+resources. While some are cached, they are cached on the basis of
+not processing individual files more than once.
+
+For the cached functions it is VERY recommended to make sure they
+are copied before any edits happen to them, as those edits may
+propagate back up into the cache. Typically the results are passed
+to `nested_combine` either immediately, or eventually after returning
+which should negate this effect.
+"""
+
+import os.path
+import sys
+from typing import Optional
+
+from sqlfluff.core.config.ini import load_ini_string
+from sqlfluff.core.config.toml import load_toml_file_config
+from sqlfluff.core.config.types import ConfigMappingType
+from sqlfluff.core.config.validate import validate_config_dict
+from sqlfluff.core.helpers.string import (
+    split_comma_separated_string,
+)
+
+if sys.version_info >= (3, 9):
```

Comment:
happy to keep this for now, but 3.8 hits EOL in October, we should consider dropping support soon after